heron/schedulers/src/java/com/twitter/heron/scheduler/kubernetes/KubernetesController.java

```diff
@@ -79,8 +83,128 @@ public boolean killTopology() {
       // Disconnect to release resources
       conn.disconnect();
     }
+  }
+
+  /**
+   * Get information about a pod
+   *
+   * @return json object for pod
+   */
+  public JsonNode getBasePod(String podId) {
+
+    String podURI = String.format(
+        "%s/api/v1/namespaces/default/pods/%s",
+        this.kubernetesURI,
+        podId);
+
+    // Get a connection
+    HttpURLConnection conn = NetworkUtils.getHttpConnection(podURI);
+    if (conn == null) {
+      LOG.log(Level.SEVERE, "Failed to find k8s deployment API");
+      return null;
+    }
+
+    try {
+      // send get request
+      if (!NetworkUtils.sendHttpGetRequest(conn)) {
+        LOG.log(Level.SEVERE, "Failed to send GET to k8s deployment API");
+        return null;
+      }
+
+      // check the response
+      boolean success = NetworkUtils.checkHttpResponseCode(conn, HttpURLConnection.HTTP_OK);
+
+      if (success) {
+        LOG.log(Level.INFO, "Pulled existing pod from k8s");
+        byte[] bytes = NetworkUtils.readHttpResponse(conn);
+
+        // read the pod definition
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode podDefinition = mapper.readTree(bytes);
+        return podDefinition;
+      } else {
+        LOG.log(Level.SEVERE, "Failure to receive existing pod from k8s");
+        return null;
+      }
+    } catch (IOException ioe) {
+      LOG.log(Level.SEVERE, "Unable to parse pod JSON");
+      return null;
+    } finally {
+      conn.disconnect();
+    }
+  }
+
+  public boolean deployContainer(String deployConf) {
+    String deploymentURI = String.format(
+        "%s/api/v1/namespaces/default/pods",
```

Comment:
variants of this uri building are repeated in this class. could you centralize this logic into a shared method or member? During initialization of this class it seems we could replace set `kubernetesURI` as `"[kubernetesURI]/api/v1/namespaces/default/pods"` which would help.